src/app/dao/FirebaseFirestoreDAO.ts

```diff
@@ -41,16 +37,37 @@ export abstract class FirebaseFirestoreDAO<T extends FirebaseJSONObject> impleme
 
     public static VERBOSE: boolean = false;
 
-    constructor(public readonly collectionName: string, protected afs: AngularFirestore) {}
+    private collection: MGPOptional<CollectionReference<T>> = MGPOptional.empty();
+
+    constructor(public readonly collectionName: string) {
+    }
+    private getCollection() {
+        if (this.collection.isAbsent()) {
+            const genericConverter: FirestoreDataConverter<T> = {
+                fromFirestore(snapshot: QueryDocumentSnapshot): T {
+                    return snapshot.data() as T;
+                },
+                toFirestore(data: PartialWithFieldValue<T>) {
+                    return data;
+                },
+            };
+            // It is necessary to not bind this.collection in the constructor,
+            // as it depends on getFirestore(), which is not instantiated when
```

Comment:
small detail but, maybe there is a better formulation to "instantiate theReturnOfAFunction()" that also might be insightful ?